core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -875,6 +901,14 @@ class KafkaZkClient private (zooKeeperClient: ZooKeeperClient, isSecure: Boolean
     retryRequestUntilConnected(deleteRequest)
   }
 
+  /**
+   * Checks the controller path existence
+   * @return true if controller exists else false
+   */
+  def controllerPathExists(): Boolean = {
```

Comment:
Rather than exposing this method, could we let callers check `getControllerId.isDefined`?